pkg/api/types.go

```diff
@@ -1802,7 +1826,25 @@ const MaxSecretSize = 1 * 1024 * 1024
 type SecretType string
 
 const (
-	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+	// SecretTypeOpaque is the default; arbitrary user-defined data
+	SecretTypeOpaque SecretType = "Opaque"
+
+	// SecretTypeServiceAccountToken contains a token that identifies a service account to the API
```

Comment:
I'm arguing that a secret-for-the-purpose-of-auth is likely to be something multiple system components produce or consume, and the fact that these are made by the service account system is a fact about their ownership, not about their type.